test/Ocelot.UnitTests/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinderTests.cs

```diff
@@ -343,22 +310,70 @@ public void can_match_down_stream_url_with_downstream_template_with_place_holder
             };
 
             this.Given(x => x.GivenIHaveAUpstreamPath("api/product/products/categories/"))
-                 .And(x => x.GivenIHaveAnUpstreamUrlTemplate("api/{finalUrlPath}/"))
-                 .When(x => x.WhenIFindTheUrlVariableNamesAndValues())
-                 .And(x => x.ThenTheTemplatesVariablesAre(expectedTemplates))
-                 .BDDfy();
+                .And(x => x.GivenIHaveAnUpstreamUrlTemplate("api/{finalUrlPath}"))
+                .When(x => x.WhenIFindTheUrlVariableNamesAndValues())
+                .And(x => x.ThenTheTemplatesVariablesAre(expectedTemplates))
+                .BDDfy();
         }
 
         [Fact]
         [Trait("Bug", "748")]
-        public void check_for_placeholder_at_end_of_template() 
+        public void check_for_placeholder_at_end_of_template()
+        {
+            var expectedTemplates = new List<PlaceholderNameAndValue> { new("{testId}", string.Empty), };
+            this.Given(x => x.GivenIHaveAUpstreamPath("/upstream/test/"))
+                .And(x => x.GivenIHaveAnUpstreamUrlTemplate("/upstream/test/{testId}"))
+                .When(x => x.WhenIFindTheUrlVariableNamesAndValues())
+                .And(x => x.ThenTheTemplatesVariablesAre(expectedTemplates))
+                .BDDfy();
+        }
+
+        [Theory]
+        [InlineData("/api/invoices/{url}-abcd", "/api/invoices/123-abcd", "{url}", "123")]
+        [InlineData("/api/invoices/{url1}-{url2}_abcd", "/api/invoices/123-456_abcd", "{url1}", "123")]
```

Comment:
Good! Thank you!
Could you write acceptance tests please?

Reply:
Let's see..